xmtp_mls/src/storage/encrypted_store/user_preferences.rs

```diff
@@ -0,0 +1,85 @@
+use crate::storage::StorageError;
+
+use super::{
+    schema::user_preferences::{self, dsl},
+    DbConnection,
+};
+use diesel::prelude::*;
+
+#[derive(Identifiable, Insertable, Queryable, Debug, Clone, PartialEq, Eq, Default)]
+#[diesel(table_name = user_preferences)]
+#[diesel(primary_key(id))]
+pub struct StoredUserPreferences {
+    /// Primary key - latest key is the "current" preference
+    pub id: Option<i32>,
+    /// Randomly generated hmac key root
+    pub hmac_key: Option<Vec<u8>>,
+}
+
+impl StoredUserPreferences {
+    pub fn load(conn: &DbConnection) -> Result<Self, StorageError> {
+        let query = dsl::user_preferences.order(dsl::id.desc()).limit(1);
+        let mut result = conn.raw_query(|conn| query.load::<StoredUserPreferences>(conn))?;
+
+        Ok(result.pop().unwrap_or_default())
+    }
+
+    pub fn set_hmac_key(conn: &DbConnection, hmac_key: Vec<u8>) -> Result<(), StorageError> {
+        let mut preferences = Self::load(conn)?;
+        preferences.id = None;
+        preferences.hmac_key = Some(hmac_key);
+
+        preferences.insert(conn)?;
+
+        Ok(())
+    }
+
+    fn insert(&self, conn: &DbConnection) -> Result<(), StorageError> {
```

Comment:
Suggestion: can't we use existing macros?

Reply:
Yeah, looks like `impl_store!` is a good use here. Thanks for the suggestion. Updated.